### PR TITLE
Fix mypy

### LINKED ISSUE #84: type test fails
```
tox -e type
```

fails with:

```
❯ tox -e type                                         ─╯
type installed:
asgiref==3.4.1,certifi==2021.5.30,charset-normalizer==2.0.3,coreapi==2.3.3,coreschema==0.0.4,Django==3.2.5,django-stubs==1.8.0,django-stubs-ext==0.2.0,djangorestframework-stubs==1.4.0,idna==3.2,itypes==1.2.0,Jinja2==3.0.1,MarkupSafe==2.0.1,mypy==0.910,mypy-extensions==0.4.3,pytz==2021.1,requests==2.26.0,sqlparse==0.4.1,toml==0.10.2,typing-extensions==3.10.0.0,uritemplate==3.0.1,urllib3==1.26.6
type run-test-pre: PYTHONHASHSEED='4185392613'
type run-test: commands[0] | mypy miqa setup.py
miqa/celery.py:3: error: Cannot find implementation or library stub for module
named "celery"
miqa/celery.py:4: error: Cannot find implementation or library stub for module
named "configurations.importer"
miqa/celery.py:4: error: Cannot find implementation or library stub for module
named "configurations"
miqa/core/tests/test_import.py:1: error: Cannot find implementation or library
stub for module named "pytest"
setup.py:3: error: Skipping analyzing "setuptools": found module but no type
hints or library stubs
miqa/settings.py:6: error: Cannot find implementation or library stub for
module named "composed_configuration"
miqa/settings.py:16: error: Cannot find implementation or library stub for
module named "composed_configuration._configuration"
miqa/settings.py:17: error: Cannot find implementation or library stub for
module named "configurations"
miqa/core/conversion/json_to_csv.py:140: error: Name "sys" is not defined
miqa/core/conversion/csv_to_json.py:37: error: Item "None" of
"Optional[Sequence[str]]" has no attribute "__iter__" (not iterable)
miqa/core/tests/test_rest.py:1: error: Cannot find implementation or library
stub for module named "pytest"
miqa/wsgi.py:3: error: Cannot find implementation or library stub for module
named "configurations.importer"
miqa/wsgi.py:3: error: Cannot find implementation or library stub for module
named "configurations"
miqa/core/models/site.py:5: error: Cannot find implementation or library stub
for module named "django_extensions.db.models"
miqa/core/models/site.py:9: error: Need type annotation for "id"
miqa/core/models/site.py:10: error: Need type annotation for "name"
miqa/core/models/site.py:11: error: Need type annotation for "creator"
miqa/core/models/session.py:5: error: Cannot find implementation or library
stub for module named "django_extensions.db.models"
miqa/core/models/session.py:9: error: Need type annotation for "id"
miqa/core/models/session.py:10: error: Need type annotation for "name"
miqa/core/models/session.py:11: error: Need type annotation for "creator"
miqa/core/models/session.py:12: error: Need type annotation for "lock_owner"
miqa/core/models/session.py:16: error: Need type annotation for "import_path"
miqa/core/models/session.py:17: error: Need type annotation for "export_path"
miqa/core/models/experiment.py:4: error: Cannot find implementation or library
stub for module named "django_extensions.db.models"
miqa/core/models/experiment.py:17: error: Need type annotation for "id"
miqa/core/models/experiment.py:18: error: Need type annotation for "name"
miqa/core/models/experiment.py:19: error: Need type annotation for "note"
miqa/core/models/experiment.py:20: error: Need type annotation for "session"
miqa/core/migrations/0002_initial.py:9: error: Cannot find implementation or
library stub for module named "django_extensions.db.fields"
miqa/core/migrations/0002_initial.py:9: error: Cannot find implementation or
library stub for module named "django_extensions"
miqa/core/migrations/0002_initial.py:9: error: Cannot find implementation or
library stub for module named "django_extensions.db"
miqa/core/management/commands/makeclient.py:2: error: Cannot find
implementation or library stub for module named "djclick"
miqa/core/management/commands/makeclient.py:2: note: See
https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
miqa/core/management/commands/makeclient.py:3: error: Cannot find
implementation or library stub for module named "oauth2_provider.models"
miqa/core/models/scan_note.py:9: error: Cannot find implementation or library
stub for module named "django_extensions.db.models"
miqa/core/models/scan_note.py:19: error: Need type annotation for "id"
miqa/core/models/scan_note.py:22: error: Need type annotation for "created"
miqa/core/models/scan_note.py:24: error: Need type annotation for "creator"
miqa/core/models/scan_note.py:25: error: Need type annotation for "initials"
miqa/core/models/scan_note.py:26: error: Need type annotation for "note"
miqa/core/models/scan_note.py:27: error: Need type annotation for "scan"
miqa/core/models/scan.py:7: error: Cannot find implementation or library stub
for module named "django_extensions.db.models"
miqa/core/models/scan.py:22: error: Need type annotation for "id"
miqa/core/models/scan.py:23: error: Need type annotation for "experiment"
miqa/core/models/scan.py:24: error: Need type annotation for "site"
miqa/core/models/scan.py:26: error: Need type annotation for "scan_id"
miqa/core/models/scan.py:27: error: Need type annotation for "scan_type"
miqa/core/models/image.py:8: error: Cannot find implementation or library stub
for module named "django_extensions.db.models"
miqa/core/models/image.py:19: error: Need type annotation for "id"
miqa/core/models/image.py:20: error: Need type annotation for "scan"
miqa/core/models/image.py:21: error: Need type annotation for "raw_path"
miqa/core/models/image.py:22: error: Need type annotation for "name"
miqa/core/models/annotation.py:47: error: Need type annotation for "id"
miqa/core/models/annotation.py:48: error: Need type annotation for "created"
miqa/core/models/annotation.py:49: error: Need type annotation for "scan"
miqa/core/models/annotation.py:50: error: Need type annotation for "creator"
miqa/core/models/annotation.py:51: error: Need type annotation for "decision"
miqa/asgi.py:3: error: Cannot find implementation or library stub for module
named "configurations.importer"
miqa/asgi.py:3: error: Cannot find implementation or library stub for module
named "configurations"
miqa/core/tasks.py:6: error: Cannot find implementation or library stub for
module named "jsonschema"
miqa/core/tasks.py:7: error: Cannot find implementation or library stub for
module named "jsonschema.exceptions"
miqa/core/tasks.py:157: error: Argument 1 to "len" has incompatible type
"Optional[Any]"; expected "Sized"
miqa/core/tests/factories.py:2: error: Cannot find implementation or library
stub for module named "factory"
miqa/core/rest/permissions.py:7: error: Module "rest_framework.views" has no
attribute "View"; maybe "_View" or "AsView"?
miqa/core/rest/permissions.py:39: error: "Model" has no attribute "session"
miqa/core/rest/permissions.py:56: error: "Model" has no attribute "session"
miqa/core/tests/conftest.py:1: error: Cannot find implementation or library
stub for module named "pytest"
miqa/core/tests/conftest.py:2: error: Cannot find implementation or library
stub for module named "pytest_factoryboy"
miqa/core/rest/site.py:1: error: Cannot find implementation or library stub
for module named "django_filters"
miqa/core/rest/session.py:3: error: Cannot find implementation or library stub
for module named "drf_yasg.utils"
miqa/core/rest/session.py:20: error: "CharField[Any, Any]" has no attribute
"field"
miqa/core/rest/scan_note.py:2: error: Cannot find implementation or library
stub for module named "django_filters"
miqa/core/rest/scan_note.py:59: error: Argument 1 of "perform_create" is
incompatible with supertype "CreateModelMixin"; supertype defines the argument
type as "BaseSerializer[Any]"
miqa/core/rest/scan_note.py:59: note: This violates the Liskov substitution
principle
miqa/core/rest/scan_note.py:59: note: See
https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
miqa/core/rest/scan_note.py:61: error: Argument 2 to "ensure_session_lock" has
incompatible type "Union[AbstractBaseUser, AnonymousUser]"; expected "User"
miqa/core/rest/image.py:4: error: Cannot find implementation or library stub
for module named "django_filters"
miqa/core/rest/annotation.py:1: error: Cannot find implementation or library
stub for module named "django_filters"
miqa/core/rest/annotation.py:35: error: Argument 1 of "perform_create" is
incompatible with supertype "CreateModelMixin"; supertype defines the argument
type as "BaseSerializer[Any]"
miqa/core/rest/annotation.py:35: note: This violates the Liskov substitution
principle
miqa/core/rest/annotation.py:35: note: See
https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
miqa/core/rest/annotation.py:37: error: Argument 2 to "ensure_session_lock"
has incompatible type "Union[AbstractBaseUser, AnonymousUser]"; expected
"User"
miqa/core/rest/scan.py:1: error: Cannot find implementation or library stub
for module named "django_filters"
miqa/core/rest/experiment.py:1: error: Cannot find implementation or library
stub for module named "django_filters"
miqa/core/management/commands/populate.py:2: error: Cannot find implementation
or library stub for module named "djclick"
miqa/urls.py:4: error: Cannot find implementation or library stub for module
named "drf_yasg"
miqa/urls.py:5: error: Cannot find implementation or library stub for module
named "drf_yasg.views"
miqa/urls.py:50: error: Cannot find implementation or library stub for module
named "debug_toolbar"
Found 84 errors in 31 files (checked 57 source files)
ERROR: InvocationError for command /home/matt/src/miqa/.tox/type/bin/mypy miqa
setup.py (exited with code 1)
________________________ summary ________________________
ERROR:   type: commands failed


```

### PR DESCRIPTION
**Pull request by annehaley**
Resolves #84

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,10 +46,12 @@ jobs:
         run: |
           tox
         env:
+          DJANGO_CONFIGURATION: DevelopmentConfiguration
```
Comment (zachmullen) [inline .github/workflows/ci.yml:49]: Why is this needed?

File: miqa/core/models/project.py [hunk 2]
```diff
@@ -69,14 +69,14 @@ def clean(self):
...
+    def get_read_permission_groups(self):
         return ['collaborator', 'tier_1_reviewer', 'tier_2_reviewer']
 
-    def get_review_permission_groups():
+    def get_review_permission_groups(self):
         return ['tier_1_reviewer', 'tier_2_reviewer']
 
     def get_user_role(self, user):
-        perm_order = Project.get_read_permission_groups()
+        perm_order = Project().get_read_permission_groups()
```
Comment (zachmullen) [inline miqa/core/models/project.py]:
```suggestion
        perm_order = self.get_read_permission_groups()
```

We should either use `self` or simply make the other methods `@staticmethods`

File: miqa/core/models/project.py [hunk 3]
```diff
@@ -108,7 +108,7 @@ def get_status(self):
         }
 
     def update_group(self, group_name, user_list):
-        if group_name not in Project.get_read_permission_groups():
+        if group_name not in Project().get_read_permission_groups():
```
Comment (zachmullen) [inline miqa/core/models/project.py]:
```suggestion
        if group_name not in self.get_read_permission_groups():
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (annehaley) [reply .github/workflows/ci.yml:49]: Without it, we get an environment error: https://github.com/OpenImaging/miqa/runs/6011730594?check_suite_focus=true

File: miqa/core/rest/experiment.py [hunk 4]
```diff
@@ -70,7 +72,7 @@ class ExperimentViewSet(ReadOnlyModelViewSet, mixins.CreateModelMixin):
     def get_queryset(self):
         projects = get_objects_for_user(
             self.request.user,
-            [f'core.{perm}' for perm in Project.get_read_permission_groups()],
+            [f'core.{perm}' for perm in Project().get_read_permission_groups()],
```
Comment (zachmullen) [inline miqa/core/rest/experiment.py:75]: We're basically calling this like a static method here, so maybe it just should be a static method? ~~Otherwise let's do~~


(I just realized this is not a model method. I think we should just make these static methods or simply settings, if they are not actually project specific.)

**Comment by zachmullen** [comment]
Whoa, does this actually get `tox -e type` to pass? If so that's hugely valuable to girder 4 projects generally.

File: miqa/core/rest/experiment.py (hunk 4 above)
Comment (annehaley) [reply miqa/core/rest/experiment.py:75]: Here, `self` is not an `Experiment` object, but an `ExperimentViewSet` object. I would try to call `self.get_object` here but `get_queryset` is called before that and would cause recursion.

File: miqa/core/models/project.py (hunk 3 above)
Comment (annehaley) [reply miqa/core/models/project.py]: Changed in [4f69033](https://github.com/OpenImaging/miqa/pull/437/commits/4f690338b966368939da389ca629d221346ca243)

File: miqa/core/models/project.py (hunk 2 above)
Comment (annehaley) [reply miqa/core/models/project.py]: Changed in [4f69033](https://github.com/OpenImaging/miqa/pull/437/commits/4f690338b966368939da389ca629d221346ca243)

**Comment by annehaley** [comment]
I didn't realize that the issue mentioned in #84 was common to other girder4 projects, but that's cool if we can share this!

**Review by zachmullen — APPROVED** [review]
